src/PKSim.Core/Services/CompoundPKContext.cs

```diff
@@ -0,0 +1,55 @@
+﻿using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Extensions;
+using PKSim.Core.Model;
+
+namespace PKSim.Core.Services
+{
+   public class CompoundPKContext
+   {
+      private readonly ICache<string, CompoundPK> _compoundPKCache;
+
+      public CompoundPKContext()
+      {
+         _compoundPKCache = new Cache<string, CompoundPK>(x => x.CompoundName, x => new CompoundPK());
+      }
+
+      public CompoundPK CompoundPKFor(string compoundName)
+      {
+         if (!_compoundPKCache.Contains(compoundName))
+            _compoundPKCache.Add(new CompoundPK { CompoundName = compoundName });
+
+         return _compoundPKCache[compoundName];
+      }
+
+      public void AddCompoundPK(CompoundPK compoundPK)
+      {
+         _compoundPKCache[compoundPK.CompoundName] = compoundPK;
+      }
+
+      /// <summary>
+      /// Initializes the context compound PK ratio parameters from the calculated simulation PK Parameters
+      /// Adds BioAvailability, AUCRatio, and CMaxRatio to the context
+      /// </summary>
+      /// <param name="populationSimulation">The simulation that contains the PK Parameters</param>
+      public void InitializeQuantityPKParametersFrom(PopulationSimulation populationSimulation)
+      {
+         populationSimulation.CompoundNames.Each(compoundName =>
```

Comment:
There is a lot of "look the same"kid of code"
What about sthg like this (
```
           var pkAnalyses = populationSimulation.PKAnalyses;
           var compoundPK = CompoundPKFor(compoundName);
            var parameters =  [
              pkAnalyses.PKParameterFor(compoundName, Bioavailability);
               pkAnalyses.PKParameterFor(compoundName,AUCRatio);
               pkAnalyses.PKParameterFor(compoundName, C_maxRatio);
           ].Where(x=>x!null);
           parameters.Each(compoundPK.AddQuantityPKParameter)


```

that way if you add one more, this is just a one line to add here
even more to

```
           var pkAnalyses = populationSimulation.PKAnalyses;
           var compoundPK = CompoundPKFor(compoundName);
            var parameters =  new [Bioavailability, AUCRatio, C_maxRatio]
.select(x=>pkAnalyses.PKParameterFor(compoundName, x)).where(x=>x!-null);
           parameters.Each(compoundPK.AddQuantityPKParameter)


```